wpilibc/src/main/native/cpp/ADIS16448_IMU.cpp

```diff
@@ -609,126 +592,95 @@ ADIS16448_IMU::~ADIS16448_IMU() {
-        imu_crc =
-            BuffToUShort(&buffer[i + 27]);  // Extract DUT CRC from data buffer
+        // Complement
+        calc_crc = ~calc_crc;
+        // Flip LSB & MSB
+        calc_crc = static_cast<uint16_t>((calc_crc << 8) | (calc_crc >> 8));
+        // Extract DUT CRC from data buffer
+        uint16_t imu_crc = BuffToUShort(&buffer[i + 27]);
 
         // Compare calculated vs read CRC. Don't update outputs or dt if CRC-16
         // is bad
         if (calc_crc == imu_crc) {
           // Timestamp is at buffer[i]
           m_dt = (buffer[i] - previous_timestamp) / 1000000.0;
-          // Split array and scale data
-          gyro_rate_x = BuffToShort(&buffer[i + 5]) * 0.04;
-          gyro_rate_y = BuffToShort(&buffer[i + 7]) * 0.04;
-          gyro_rate_z = BuffToShort(&buffer[i + 9]) * 0.04;
-          accel_x = BuffToShort(&buffer[i + 11]) * 0.833;
-          accel_y = BuffToShort(&buffer[i + 13]) * 0.833;
-          accel_z = BuffToShort(&buffer[i + 15]) * 0.833;
-          mag_x = BuffToShort(&buffer[i + 17]) * 0.1429;
-          mag_y = BuffToShort(&buffer[i + 19]) * 0.1429;
-          mag_z = BuffToShort(&buffer[i + 21]) * 0.1429;
-          baro = BuffToShort(&buffer[i + 23]) * 0.02;
-          temp = BuffToShort(&buffer[i + 25]) * 0.07386 + 31.0;
+          // Scale sensor data
+          double gyro_rate_x = BuffToShort(&buffer[i + 5]) * 0.04;
+          double gyro_rate_y = BuffToShort(&buffer[i + 7]) * 0.04;
+          double gyro_rate_z = BuffToShort(&buffer[i + 9]) * 0.04;
+          double accel_x = BuffToShort(&buffer[i + 11]) * 0.833;
+          double accel_y = BuffToShort(&buffer[i + 13]) * 0.833;
+          double accel_z = BuffToShort(&buffer[i + 15]) * 0.833;
+          double mag_x = BuffToShort(&buffer[i + 17]) * 0.1429;
+          double mag_y = BuffToShort(&buffer[i + 19]) * 0.1429;
+          double mag_z = BuffToShort(&buffer[i + 21]) * 0.1429;
+          double baro = BuffToShort(&buffer[i + 23]) * 0.02;
+          double temp = BuffToShort(&buffer[i + 25]) * 0.07386 + 31.0;
 
           // Convert scaled sensor data to SI units
-          gyro_rate_x_si = gyro_rate_x * deg_to_rad;
-          gyro_rate_y_si = gyro_rate_y * deg_to_rad;
-          // gyro_rate_z_si = gyro_rate_z * deg_to_rad;
-          accel_x_si = accel_x * grav;
-          accel_y_si = accel_y * grav;
-          accel_z_si = accel_z * grav;
+          double gyro_rate_x_si = gyro_rate_x * deg_to_rad;
+          double gyro_rate_y_si = gyro_rate_y * deg_to_rad;
+          // double gyro_rate_z_si = gyro_rate_z * deg_to_rad;
+          double accel_x_si = accel_x * grav;
+          double accel_y_si = accel_y * grav;
+          double accel_z_si = accel_z * grav;
           // Store timestamp for next iteration
           previous_timestamp = buffer[i];
           // Calculate alpha for use with the complementary filter
-          m_alpha = m_tau / (m_tau + m_dt);
+          m_alpha = 0.5 / (0.5 + m_dt);
```

Comment:
It'd be nice to still have a constant for tau so that we don't have a random magic number.